.github/workflows/stack.yml

```diff
@@ -0,0 +1,103 @@
+name: Stack
+
+on:
+  push:
+    paths-ignore:
+    - 'demo/**'
+    - 'docker/**'
+    - 'Dockerfile'
+    - '.dockerignore'
+    - '.ghci'
+    - '.gitignore'
+    - 'images/**'
+    - 'notebooks/**'
+    - 'README.md'
+    - 'hie.yaml'
+  pull_request:
+    paths-ignore:
+    - 'demo/**'
+    - 'docker/**'
+    - 'Dockerfile'
+    - '.dockerignore'
+    - '.ghci'
+    - '.gitignore'
+    - 'images/**'
+    - 'notebooks/**'
+    - 'README.md'
+    - 'hie.yaml'
+
+jobs:
+  linux:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        stack-yaml:
+        # - 'stack/stack-8.2.yaml'
+        # - 'stack/stack-8.4.yaml'
+        # - 'stack/stack-8.6.yaml'
+        - 'stack/stack-8.8.yaml'
+        - 'stack/stack-8.10.yaml'
+        - 'stack/stack-9.0.yaml'
+        - 'stack/stack-9.2.yaml'
+        - 'stack/stack-9.4.yaml'
+        - 'stack/stack-9.6.yaml'
+        - 'stack.yaml'
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: cachix/install-nix-action@v22
+      with:
+        install_url: https://releases.nixos.org/nix/nix-2.13.3/install
+
+    - uses: haskell/actions/setup@v2
+      name: Setup Haskell Stack
+      with:
+        enable-stack: true
+        stack-version: "latest"
+
+    - uses: actions/cache@v3
+      name: Cache ~/.stack
+      with:
+        path: ~/.stack
+        key: ${{ runner.os }}-${{ matrix.stack-yaml }}-${{ hashFiles(format('./{0}', matrix.stack-yaml)) }}
+
+    - name: Build
+      run: |
+        export "NIX_PATH=nixpkgs=$(nix run .#print-nixpkgs-master)"
+        stack build --nix --stack-yaml ${{matrix.stack-yaml}}
```

Comment:
It's a bit janky but I think we want to copy the `stack-<version>.yaml` to the toplevel as `stack.yaml`, because it's what our Stack integration looks for during the integration test.

Reply:
Done, though it doesn't seem to have helped with the current problem -- let's discuss below